### ISSUE TITLE
Empty NotificationConfiguration for PutBucketNotificationConfigurationRequest produces an empty request body

### ISSUE THREAD

**Issue by tinkyholloway** [aws/aws-sdk-cpp#733]
Apologies if I am doing something wrong.

The documentation says that in order to deactivate bucket notifications then a PUT with an emptry NotificationConfiguration should be made.

```
Aws::S3::Model::NotificationConfiguration notificationConfiguration;
Aws::S3::Model::PutBucketNotificationConfigurationRequest putBucketNotificationConfigurationRequest;                                                                                                                                     
putBucketNotificationConfigurationRequest.SetBucket(BUCKET);
putBucketNotificationConfigurationRequest.SetNotificationConfiguration(notificationConfiguration);
```

Instead of producing a request body with `<NotificationConfiguration></NotificationConfiguration>`a request with an empty body is generated and AWS responds:

```
<?xml version="1.0" encoding="UTF-8"?>
<Error>
    <Code>MissingRequestBodyError</Code>
    <Message>Request Body is empty</Message>
    <RequestId>3D62792603442F6C</RequestId>
    <HostId>RTb6Aw2yYAWw4ViJQQUUqs6kVBpi0F2XRdvlyGbgTmmyChZ4zciISaJHQtiJAsSIGx+/0dmgpec=</HostId>
</Error>
```

If it turns out that this is a bug instead of my user error, is it possible to overwrite the request body before the request is submitted?

**Comment by marcomagdy** [comment]
Probably a bug on our end. I'm looking into it.

**Comment by marcomagdy** [comment]
Merging a fix today

**Comment by marcomagdy** [comment]
Fixed in v1.3.30
